doc/icache.rst

```diff
@@ -269,3 +269,9 @@ While doing this, the cache behaves as if ``icache_enable_i`` is false and will
 
 The ``busy_o`` signal is guaranteed to be high while the cache is invalidating its internal memories or whenever it has a pending fetch on the instruction bus.
 When the ``busy_o`` signal is low, it is safe to clock gate the cache.
+
+The cache doesn't have circuitry to avoid inconsistent multi-way hits.
+As such, the core must never fetch from an address with the cache enabled after modifying the data at that address, without first starting a cache invalidation.
+
+.. note::
+   This is a constraint on *software*, not just on the core.
```

Comment:
Yep looks good to me - could reference the "Zifencei" of the unpriv spec for background if you wanted.

Reply:
I've just looked at that chapter and can't find the bit that says "valid code must issue a fence.i instruction before executing from code memory that might have been touched". It sounds like something that might not belong in the ISA (since it's only required by some cache architectures). Am I looking in the wrong place?

Reply:
I think the closest thing is the sentence:
" RISC-V does not guarantee that stores to instruction memory will be made visible to instruction fetches on a RISC-V hart until that hart executes a FENCE.I instruction."

Reply:
Yep, ok. Of course, that doesn't say "and you might get random garbage rather than *either* version of the instruction" :-)

I guess I'll just leave it as is for now. Presumably we'll have some sort of platform document at some point, which will say this sort of thing (in OT- and/or Ibex-land).